executor/src/witgen/generator.rs

```diff
@@ -106,27 +125,26 @@ where
                     }
                 };
 
-                progress |= self.handle_eval_result(result);
+                progress |= self.handle_eval_result(result, strategy);
             }
 
-            if self.query_callback.is_some() {
+            if self.query_callback.is_some() && strategy == SolvingStrategy::SingleVariableAffine {
                 for column in self.fixed_data.witness_cols() {
-                    // TODO we should actually query even if it is already known, to check
-                    // if the value would be different.
                     if !self.has_known_next_value(column.poly.poly_id() as usize)
                         && column.query.is_some()
                     {
                         let result = self.process_witness_query(&column);
-                        progress |= self.handle_eval_result(result);
+                        progress |= self.handle_eval_result(result, strategy);
                     }
                 }
             }
 
-            if !progress {
-                break;
-            }
-            if self.next.iter().all(|v| v.is_some()) {
```

Comment:
It seems like this exit condition (when all v are set) is missing in the new code.

Reply:
Yes, this is intentional to add further checks.